src/zenml/materializers/base_materializer.py

```diff
@@ -260,6 +262,38 @@ def save(self, data: Any) -> None:
             )
             self.handle_return(data)
 
+    def extract_metadata(self, data: Any) -> Dict[str, "MetadataType"]:
+        """Extract metadata from the given data.
+
+        This metadata will be tracked and displayed alongside the artifact.
+
+        Args:
+            data: The data to extract metadata from.
+
+        Returns:
+            A dictionary of metadata.
+
+        Raises:
+            TypeError: If the data is not of the correct type.
+        """
+        from zenml.metadata.metadata_types import StorageSize
+
+        data_type = type(data)
+        if not self._can_handle_type(data_type):
+            raise TypeError(
+                f"Unable to extract metadata from {data_type}. "
+                f"{self.__class__.__name__} can only write the following "
+                f"types: {self.ASSOCIATED_TYPES}."
+            )
+
+        metadata: Dict[str, "MetadataType"] = {
+            "runtime_data_type": DType(type(data))
```

Comment:
```suggestion
            "runtime_data_type": DType(data_type)
```

Can we also just name this property `data_type`, why is the runtime prefix necessary?

Reply:
Still undecided about this

Reply:
Will remove it since it is duplicated from `data_type`